pkg/didcomm/common/messenger/messenger.go

```diff
@@ -0,0 +1,36 @@
+/*
+Copyright SecureKey Technologies Inc. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package messenger
+
+import "github.com/hyperledger/aries-framework-go/pkg/didcomm/common/service"
+
+// The messenger package is responsible for the handling of communication between agents.
+// It takes care of threading, timing, etc.
+// Each message that we are going to send should be DIDCommMsg type.
+// e.g we have type model.Ack{Type: "type", ID: "id"}
+// it should be converted to: map[@id:id @type:type]
+// NOTE: The package modifies message data by JSON tag name according to aries-rfcs.
+//       Fields like @id, ~thread , etc. are redundant and may be rewritten.
+
+// Client provides methods for the communication
+type Client interface {
+	// HandleMessage receives incoming message and prepares data needed for the communication according to the options.
+	// Options should be provided by the service.
+	HandleMessage(msg service.DIDCommMsg, myDID, theirDID string, opts ...interface{})
```

Comment:
- if you plan to expose this interface through context, then `HandleMessage()` should be moved out of this interface since it doesn't make sense to expose it. Services/Clients shouldn't call this function explicitly.
My suggestion to split this package and expose only Send/Reply in client interface.
- Name `Client` doesn't look good if it is going to be exposed through context
- Should get rid of current outbound dispatcher once this interface is available in context

Reply:
I will remove HandleMessage from the interface. What about the name? Suggestions?

Reply:
@sudeshrshetty @soluchok @rolsonquadras  what's the status on this comment?

Reply:
`Messenger` should be the right name for this interface, but `messenger.Messenger` stutters.
Should we think about moving this interface to `dispatcher` package?

Reply:
Done!